pkg/dns/dns.go

```diff
@@ -48,28 +49,38 @@ const (
 
 	// Resync period for the kube controller loop.
 	resyncPeriod = 30 * time.Minute
+
+	// Duration for which the TTL cache should hold the node.
+	nodeCacheTTL = 180 * time.Second
 )
 
 type KubeDNS struct {
 	kubeClient *kclient.Client
 	// DNS domain name.
 	domain string
+	// Map of federation names that the cluster in which this kube-dns is running belongs to, to
+	// the corresponding domain names.
+	federations map[string]string
```

Comment:
No. There is a little subtlety here. `federationsVar` is a struct that satisfies the `Value` interface and wraps around the `nameDomainMap` map in which we are really interested. We ultimately want to have the command line flag values in the `federations` map here because that's all we care about, the map. So we alias the `federationsVar.nameDomainMap` to point to `federations`, so when pflag deserializes the input command line values by calling the `federationVars.Set()` method, the key-values are copied to `federationsVar.nameDomainMap` and hence `federations` map here.